src/scripts/location-select.js

```diff
@@ -99,27 +100,105 @@ function goodForecasts(acc, forecast) {
     return acc.concat(forecast);
 }
 
-// On load, fetch cities.json for input field, render as options.
-fetch('/cities.json').then((res) => res.json()).then((cities) => {
-    $cities.innerHTML = cities.map((entry) => {
-        return `<option>${formatLabel(entry)}</option>`;
+/**
+ * Moves the tabindex between buttons in the flyout using the up/down arrow keys.
+ * 
+ * @param {KeyboardEvent} ev - Native DOM keyboard event 
+ * @returns {Element|undefined} The next element to focus
+ */
+function keyboardTraverse(ev) {
+    if (!ARROW_KEYS.includes(ev.key)) return;
+    ev.stopPropagation();
+    const visible = [...$locations.querySelectorAll('button:not([style*="none"])')];
+    const current = visible.findIndex(($btn) => $btn.tabIndex === 0);
+    if (!~current) return updateTabIndex(visible.at(0));
+    const direction = (ARROW_KEYS.indexOf(ev.key) * 2) - 1;
+    const next = (current + direction + visible.length) % visible.length;
+    return updateTabIndex(visible.at(next));
+}
+
+/**
+ * Updates the tabIndex of all buttons in the flyout.
+ * 
+ * @param {Element} $current - The element meant to be newly tabbable
+ * @returns {Element} - The element meant to be newly tabbable
+ */
+function updateTabIndex($current) {
+    if (!$current) return;
+    [...$locations.children].forEach(($btn) => {
+        $btn.tabIndex = $btn === $current ? 0 : -1;
+    });
+    $current.scrollIntoView({ block: 'nearest' });
+    return $current;
+}
+
+// Show or hide the flyout depending on if the input has a value.
+$input.addEventListener('click', (ev) => 
```

Comment:
it's funny to see all these top level event listeners being added back in lol

Reply:
Yeah, they're necessary again since `<datalist/>` isn't what I expected.